HIF.md

```diff
@@ -0,0 +1,739 @@
+
+### `0x21 REQ_RADIO_LIST`
+
+Request the list of radio configuration supported by the device, typically
+issued by the host during a startup sequence before selecting a PHY
+configuration. The RCP will answer with a series of [`CNF_RADIO_LIST`][rf-list]
+commands.
+
+Body of this command is empty.
+
+### `0x22 CNF_RADIO_LIST`
+
+List of radio configuration supported by the RCP, as configured during project
+generation. Entries are defined in groups of mode-switch compatible PHYs. For
+OFDM configurations, all MCS are defined in the same entry.
+
+ - `uint8_t entry_size`  
+    Size of a radio configuration entry in bytes. This is meant to support API
+    evolution.
+
+ - `bool list_end`  
+    If not set, the list will continue in another [`CNF_RADIO_LIST`][rf-list].
+
+ - `uint8_t count`  
+    Number of radio configuration entries in next field.
+
+ - `struct rf_config[]`  
+    - `uint16_t flags`
+        - `0x0001`: If set, this entry is in same group than the previous.
+        - `0x01FE`: Bitfield of supported OFDM MCS (from MCS0 to MCS11).
+    - `uint8_t rail_phy_mode_id`  
+       Wi-SUN PhyModeId, but for OFDM, only MCS0 exists.
+    - `uint32_t chan_f0`  
+       Frequency of the first channel in Hz.
+    - `uint32_t chan_spacing`  
+       Channel spacing in Hz.
+    - `uint16_t chan_count`  
+       Number of channels without holes in the spectrum.
+    - `uint16_t sensitivity` (API >= 2.4.0)  
+       Minimum RX sensitivity in dBm for this PHY.
+
+### `0x23 SET_RADIO`
+
+Configure the radio parameters.
+
+ - `uint8_t index`  
+    Index in the `rf_config` list.
+
+ - `uint8_t mcs`  
+    MCS to be used if `index` points to an OFDM modulation.
+
+ - `bool enable_ms` (API > 2.0.1)  
+    Enable mode switch in RX and TX, using all the PHY configurations available
+    in the same group as the selected PHY. Mode switch is enabled at PHY level
+    (PHR) and MAC level (MAC command frame). See [`REQ_DATA_TX`][tx-req] for TX
+    parameters.
+
+### `0x24 SET_RADIO_REGULATION`
+
+Enable specific radio regulation rules. Most regulations only make sense with
+specific channel configurations.
+
+ - `uint32_t value`  
+    - `0`: None (disabled, default)
+    - `2`: Japan ([ARIB][arib])
+    - `5`: India ([WPC][wpc])  
+
+[arib]: https://www.arib.or.jp/
+[wpc]:  https://dot.gov.in/spectrum-management/2457
+
+### `0x25 SET_RADIO_TX_POWER`
+
+ - `int8_t tx_power_dbm`  
+    Maximum transmission power in dBm. The RCP may utilize a lower value based
+    on internal decision making or hardware limitations but will never exceed
+    the given value. The default value is 14dBm.
+
+## Frequency Hopping (FHSS) configuration
+
+Several schemes are used in Wi-SUN to transmit or receive packets using
+frequency hopping. These are independently defined using a set of parameters,
+and may differ between transmission and reception. The following table
+summarizes which commands configure the various FHSS parameters. Note that
```

Comment:
If we are talking about pure configuration, only the `SET_*` commands should be documented here.
Maybe writing about which `SET` cmd should be called to **enable** the various RX/TX capabilities would be better?

Then we can have a separate section about Wi-SUN-specific timings filled up by the RCP on `REQ_DATA_TX` (like we already have, but I think a new section/subsection is necessary).

Reply:
The goal of this section is to summarize all the schemes available for RX/TX and all that is necessary to make them work with the whole HIF API, not just the `SET_FHSS` commands. This paragraph should be referenced in the same way that the security section is referenced throughout the data TX/RX section (on that note maybe I'm missing a link somewhere in `REQ_DATA_TX`).

Wi-SUN is arguably quite convoluted when it comes to this so I think it is sane putting everything in a big picture.